### PR TITLE
Update eslint and node-sass to latest versions

### PR DESCRIPTION
**Pull request by Lukewh**
## QA

- Pull the branch
- Ensure your dev environment still works ok
- Ensure building, linting and testing still work ok

### CODE REVIEW

File: static/js/base/ga.js [hunk 1]
```diff
@@ -72,7 +72,9 @@ if (typeof dataLayer !== "undefined") {
       return;
     }
 
-    for (let key in events) {
+    let key;
```
Comment (Lukewh) [inline static/js/base/ga.js]: I can't find a rule that allows this - it seems to be a recommended pattern

**Comment by bartaz** [comment]
Probably need to update eslint plugins as well:
`warning " > eslint-plugin-react@7.10.0" has incorrect peer dependency "eslint@^3.0.0 || ^4.0.0 || ^5.0.0"`

File: static/js/base/ga.js (hunk 1 above)
Comment (bartaz) [reply static/js/base/ga.js]: I think is a eslint + babel-eslint bug: https://github.com/eslint/eslint/issues/12117

Maybe we should update all eslint related deps.

**Comment by bartaz** [comment]
I have some problems running prettier. Maybe prettier plugin needs updating as well?

**Pull request closed without merging**